### PR TITLE
Fix compile warnings when using clang-15

### LINKED ISSUE #5717: Build boost fail when using clang15 and libc++15
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Build tiflash using clang15 and libc++15

### 2. What did you expect to see? (Required)

Build pass

### 3. What did you see instead (Required)

tiflash/contrib/boost/boost/container_hash/hash.hpp:131:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^~~~~~~~~~~~~~
                                __unary_function
/usr/lib/llvm-15/bin/../include/c++/v1/__functional/unary_function.h:46:1: note: '__unary_function' declared here
using __unary_function = __unary_function_keep_layout_base<_Arg, _Result>;
^

### 4. What is your TiFlash version? (Required)

master 61e436be45a9df7daa3869fd4794cd155ff9896b

`std::unary_function` is deprecated in C++11 and removed in C++17. From [Boost Macro Reference](https://www.boost.org/doc/libs/1_76_0/libs/config/doc/html/boost_config/boost_macro_reference.html#boost_config.boost_macro_reference.macros_that_describe_features_that_have_been_removed_from_the_standard_) 
 set `BOOST_NO_CXX98_FUNCTION_BASE` could solve this issue.

### PR DESCRIPTION
**Pull request by fuzhe1989**
### What problem does this PR solve?

Issue Number: close #5717

Problem Summary:

### What is changed and how it works?

1. Suppress clang warnings `-Wunused-but-set-variable` and `-Wdeprecated-builtins` for boost_internal.
2. Wrap boost headers for suppressing clang warnings.
3.  Remove some unused variables (Dictionary).
4. Change `std::uniform_int_distribution<char>` to `std::uniform_int_distribution<uint8_t>`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: libs/libcommon/include/common/types.h [hunk 1]
```diff
@@ -18,21 +18,9 @@
 #include <string>
 
 /// import UInt128 and UInt256
+#include <boost_wrapper/cpp_int.h>
 #include <common/UInt128.h>
```
Comment (gengliqi) [inline libs/libcommon/include/common/types.h:22]:
```suggestion
/// import UInt128 and UInt256
#include <common/UInt128.h>
#include <boost_wrapper/cpp_int.h>
```

File: libs/libcommon/include/common/types.h [hunk 2]
```diff
@@ -18,21 +18,9 @@
 #include <string>
 
 /// import UInt128 and UInt256
+#include <boost_wrapper/cpp_int.h>
```
Comment (fuzhe1989) [inline libs/libcommon/include/common/types.h:21]:
```suggestion
#include <boost_wrapper/cpp_int.h>
/// import UInt128 and UInt256
```

**Review by LittleFall — DISMISSED** [review]

File: .clang-tidy [hunk 3]
```diff
@@ -139,7 +139,6 @@ Checks: '-*,
     clang-analyzer-core.uninitialized.UndefReturn,
     clang-analyzer-cplusplus.InnerPointer,
     clang-analyzer-cplusplus.NewDelete,
-    clang-analyzer-cplusplus.NewDeleteLeaks,
```
Comment (JaySon-Huang) [inline .clang-tidy:142]: We need to add this after upgrade the CI toolchain to clang-15

Comment (JaySon-Huang) [reply .clang-tidy:142]: a false positive in boost::any https://bugs.llvm.org/show_bug.cgi?id=41141

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Review by gengliqi — APPROVED** [review]
lgtm2

**Pull request merged**